src/engine/strat_engine/liminal.rs

```diff
@@ -529,6 +529,74 @@ impl LiminalDevices {
             .is_none());
     }
 
+    /// Unlock the liminal encrypted devices that correspond to the given pool UUID.
+    pub fn unlock_pool(&mut self, pool_uuid: PoolUuid) -> StratisResult<Vec<DevUuid>> {
+        fn handle_luks(luks_info: &LLuksInfo) -> StratisResult<()> {
+            if let Some(mut handle) = CryptHandle::setup(&luks_info.ids.devnode)? {
+                handle.activate()?;
+                Ok(())
+            } else {
+                Err(StratisError::Engine(
+                    ErrorEnum::Invalid,
+                    format!(
+                        "Block device {} does not appear to be formatted with
+                        the proper Stratis LUKS2 metadata.",
+                        luks_info.ids.devnode.display(),
+                    ),
+                ))
+            }
+        }
+
+        let unlocked = match self.errored_pool_devices.get(&pool_uuid) {
+            Some(map) => {
+                let mut unlocked = Vec::new();
+                for (dev_uuid, info) in map.iter() {
+                    match info {
+                        LInfo::Stratis(_) => (),
+                        LInfo::Luks(ref luks_info) => {
+                            match handle_luks(luks_info) {
+                                Ok(()) => unlocked.push(*dev_uuid),
+                                Err(e) => return Err(e),
+                            };
+                        }
+                    };
+                }
+                unlocked
+            }
+            None => {
+                return Err(StratisError::Engine(
+                    ErrorEnum::Error,
+                    format!("No pool with UUID {} exists", pool_uuid.to_simple_ref()),
+                ))
+            }
+        };
+
+        Ok(unlocked)
+    }
+
+    /// Get a list of pool UUIDs from all of the LUKS2 devices that are currently
+    /// locked in the set of pools that are not yet set up.
+    pub fn locked_pool_uuids(&self) -> Vec<PoolUuid> {
+        self.errored_pool_devices
+            .iter()
+            .filter_map(|(pool_uuid, map)| {
+                let has_locked = map.iter().fold(false, |has_locked, (_, info)| {
```

Comment:
```fold``` works here, but ```any``` is more direct and a bit easier to understand. It's used elsewhere, probably in ```setup_pool```.

Reply:
I think that at some point we might want to establish a purpose-built data structure for the values of ```errored_pool_devices```, but that's not something we need to worry about now.

Reply:
Resolved.